### ISSUE TITLE
Stem is unmaintained.

### ISSUE THREAD

**Issue by kkarhan** [torproject/stem#154]
As hinted by me [in this issue re: nyx](https://github.com/torproject/nyx/issues/68) and confirmed by @atagar as [per his comment](https://github.com/torproject/nyx/issues/68#issuecomment-1932907503) both stem and nyx are unmaintained.

**Comment by gk-tpo** [comment]
So, here is what ideally happens from a Tor Project side: Both nyx and stem should be archived on Github. Then we'd take patches against stem via our maintenance branch on Gitlab: https://gitlab.torproject.org/tpo/network-health/stem/-/tree/maint?ref_type=heads. The flow for stem right now (which is okay for us) is explained on https://stem.torproject.org/.

For nyx we don't plan doing any releases and for stem we do bugfix/maintenance releases until we transitioned away from it our tools and the new world with arti (https://gitlab.torproject.org/tpo/core/arti) is a reality.

However, the call what to do with both projects, nyx and stem, on Github is something @atagar has to make.

**Comment by emdee-is** [comment]
@gk-tpo Who has write privs on the repo besides @atagar ?

**Comment by atagar** [comment]
I don't interact with Tor's Gitlab. Moving Stem's canonical home there would remove the only person who replies to tickets or merges patches so that probably not a good idea. ;P

**Comment by emdee-is** [comment]
The project should migrate away from setup.py

You can use these as starting points:
[setup.cfg.txt](https://github.com/torproject/stem/files/14351969/setup.cfg.txt)
[pyproject.toml.txt](https://github.com/torproject/stem/files/14351967/pyproject.toml.txt)

**Comment by atagar** [comment]
> @atagar are you the only one who has write privs on this github repo?

No. GK, Juga, Silvia, Alex (ahf), Nick, George (asn), Mike, and David Goulet do too.

**Comment by emdee-is** [comment]
Thanks; I looked at the 1.8.3 gitlab code and there are some small changes from cryprtography renames and some data updates but the unit tests do not run cleanly for me.

[stem.github.diff.txt](https://github.com/torproject/stem/files/14352397/stem.github.diff.txt)

Now we have 2 stems: now what?

**Comment by hendursaga** [comment]
Just a suggestion, but perhaps pin this issue?

**Comment by atagar** [comment]
I'm sorry. On reflection Georg is right, this project should move to Gitlab.

Pinned this issue. Thanks for suggesting that, hendursaga.

**Comment by Vizonex** [comment]
Sorry if I am asking about this library over a year after this issue has been posted I thought making an asynchronous version called `aiostem`  which would be an asynchronous version with updated features and python2 being fully dropped out of the equation but I didn't want to utilize any obsolete techniques and arti is still in it's beta stage with no good relative ties to python or even asyncio yet. Before I attempt to bite the bullet on this project idea that I had in mind, should I try using the new arti-rust library and bind it with pyo3 or is there any other suggestions that others might have in mind for me?
